apis/v1/gateway_types.go

```diff
@@ -32,6 +32,10 @@ import (
 
 // Gateway represents an instance of a service-traffic handling infrastructure
 // by binding Listeners to a set of IP addresses.
+//
+// It is recommended that the Gateway name field be limited to a maximum length
+// of 63 characters. Gateway names may be used in label values on generated
```

Comment:
It seems like 63 is too long if 63 is the max length of some of the values we're trying to populate. Ideally wouldn't you want to translate `foo-gateway` to a generated `contour-foo-gateway` or similar? I'm also not sure if this belongs in the API spec or should just be in more user-facing guidelines documentation for using the API.

Reply:
63 is the max label value length, this shouldn't have any immediate bearing on generated resource names

Reply:
I can move this to another guide instead though if that is preferred, maybe better here: https://gateway-api.sigs.k8s.io/api-types/gateway/

Reply:
>63 is the max label value length, this shouldn't have any immediate bearing on generated resource names

🤦  did not read this closely enough, nevermind.

>I can move this to another guide instead

We don't really have any strong guidance here as far as where user-facing tips like this belongs. Generally we reserve RFC language like SHOULD and MUST in the API Spec for _implementation_ guidance, but it also feels weird to have "It is recommended" instead of "SHOULD" here. I can't really think of better solutions here, but interested in what @youngnick thinks.

Reply:
Yeah, this is a good use case for SHOULD - in that we would prefer implementations be aware of the complexities here as well.

Let's change "It is recommended..." to "The Gateway name field SHOULD be no longer than 63 characters."